### PR TITLE
minor trivial fixes set

### PR DESCRIPTION
**Pull request by alexrayne**
here provided a set of fixes tat i need to apply to build native and cooja motes with WERROR strict.

also included fix on Issue #1249

### CODE REVIEW

File: arch/cpu/native/Makefile.native [hunk 1]
```diff
@@ -15,7 +15,10 @@ NM       ?= nm
 OBJCOPY  ?= objcopy
 STRIP    ?= strip
 ifeq ($(WERROR),1)
-CFLAGSWERROR=-Werror
+CFLAGSWERROR +=-Werror 
+CFLAGSWERROR +=-Wno-error=unused-function -Wno-unused-function
+# printf( (char*)pfmt ) - here fmt is variable, may warns on modern compiler
```
Comment (atiselsts) [inline arch/cpu/native/Makefile.native:20]: Please remove the commented out lines, then I can merge the PR. 
There is no need to close and resubmit, just add the change in a new commit and `git rebase -i contiki-orig/develop`. Then select "fixup" for the new commit to include it in an existing one, and then `git push -f origin`
See https://github.com/contiki-os/contiki/wiki/Code-Contributions#how-to-contribute-code

File: arch/platform/cooja/Makefile.cooja [hunk 2]
```diff
@@ -72,5 +72,8 @@ CLEAN += COOJA.log
 CFLAGSNO = $(EXTRA_CC_ARGS) -Wall -g -I/usr/local/include -DCLASSNAME=$(CLASSNAME)
 ifeq ($(WERROR),1)
 CFLAGSNO += -Werror
+CFLAGSNO += -Wno-error=unused-function -Wno-unused-function 
+# printf( (char*)pfmt ) - here fmt is variable, may warns on modern compiler
```
Comment (atiselsts) [inline arch/platform/cooja/Makefile.cooja:76]: Same - no need for the commented out lines

**Comment by atiselsts** [comment]
Hmm, I wonder why some checks are in "Waiting" phase.

Could you please address my comments, rebase and push? See what happens.

**Comment by g-oikonomou** [comment]
> Hmm, I wonder why some checks are in "Waiting" phase.

Since we merged the github actions PR, github put all respective tests in waiting phase. For older PRs (opened before github actions), there are no github actions tests specified. These "waiting" states will go away with a rebase (or my merging develop into the PR branch).

**Comment by alexrayne** [comment]
luck of motivation

**Pull request closed without merging**